corehq/sql_accessors/sql_templates/get_case_ids_in_domain_1.sql

```diff
@@ -0,0 +1,31 @@
+DROP FUNCTION IF EXISTS get_case_ids_in_domain(TEXT, TEXT, TEXT[], BOOLEAN);
+
+CREATE FUNCTION get_case_ids_in_domain(
+    domain_name TEXT,
+    case_type TEXT DEFAULT NULL,
+    owner_ids TEXT[] DEFAULT NULL,
+    p_closed BOOLEAN DEFAULT NULL
+) RETURNS TABLE (case_id VARCHAR(255)) AS $$
+DECLARE
+    query_expr      TEXT := 'SELECT case_id FROM form_processor_commcarecasesql WHERE domain = $1';
+    type_filter     TEXT := ' AND type = $2';
+    owner_filter    TEXT := ' AND owner_id = ANY($3)';
+    closed_filter   TEXT := ' AND closed = $4';
+BEGIN
+    IF case_type <> '' THEN
```

Comment:
interesting, thanks.

On Tue, May 31, 2016 at 12:01 PM, Simon Kelly <notifications@github.com>
wrote:

> In corehq/sql_accessors/sql_templates/get_case_ids_in_domain_1.sql
> <https://github.com/dimagi/commcare-hq/pull/11841#discussion_r65153969>:
>
> > @@ -0,0 +1,31 @@
> > +DROP FUNCTION IF EXISTS get_case_ids_in_domain(TEXT, TEXT, TEXT[], BOOLEAN);
> > +
> > +CREATE FUNCTION get_case_ids_in_domain(
> > +    domain_name TEXT,
> > +    case_type TEXT DEFAULT NULL,
> > +    owner_ids TEXT[] DEFAULT NULL,
> > +    p_closed BOOLEAN DEFAULT NULL
> > +) RETURNS TABLE (case_id VARCHAR(255)) AS $$
> > +DECLARE
> > +    query_expr      TEXT := 'SELECT case_id FROM form_processor_commcarecasesql WHERE domain = $1';
> > +    type_filter     TEXT := ' AND type = $2';
> > +    owner_filter    TEXT := ' AND owner_id = ANY($3)';
> > +    closed_filter   TEXT := ' AND closed = $4';
> > +BEGIN
> > +    IF case_type <> '' THEN
>
> No it's not but this that case_type is neither NULL or emtpy:
> http://stackoverflow.com/a/23767625
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/dimagi/commcare-hq/pull/11841/files/3c3ab8c0b643e7d588d27df45e6b01c899e1f4cd#r65153969>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAED-1Plsr5rwHA3_VUm93Q-mu7w88E-ks5qHAbqgaJpZM4IqAfd>
> .
>